solidity/random-beacon/test/RandomBeacon.Authorization.test.ts

```diff
@@ -0,0 +1,2949 @@
+          )
+          await randomBeacon.notifyDkgTimeout()
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        context("when sortition pool was not yet updated", () => {
+          it("should return false", async () => {
+            expect(await randomBeacon.isOperatorUpToDate(operator.address)).to
+              .be.false
+          })
+        })
+
+        context("when the sortition pool was updated", () => {
+          it("should return true", async () => {
+            await randomBeacon.updateOperatorStatus(operator.address)
+            expect(await randomBeacon.isOperatorUpToDate(operator.address)).to
+              .be.true
+          })
+        })
+      })
+    })
+  })
+
+  // Testing final states for scenarios when functions are invoked one after
+  // another. Operator is known and registered in the sortition pool.
+  context("mixed interactions", () => {
+    let initialIncrease
+
+    before(async () => {
+      await createSnapshot()
+
+      await randomBeacon
+        .connect(stakingProvider)
+        .registerOperator(operator.address)
+
+      // Authorized almost the entire staked amount but leave some margin for
+      // authorization increase.
+      initialIncrease = stakedAmount.sub(to1e18(20000))
+      await staking
+        .connect(authorizer)
+        .increaseAuthorization(
+          stakingProvider.address,
+          randomBeacon.address,
+          initialIncrease
+        )
+      await randomBeacon.connect(operator).joinSortitionPool()
+    })
+
+    after(async () => {
+      await restoreSnapshot()
+    })
+
+    // Invoke `increaseAuthorization` after `increaseAuthorization`.
+    describe("authorizationIncreased -> authorizationIncreased", () => {
+      let secondIncrease
+
+      before(async () => {
+        await createSnapshot()
+
+        secondIncrease = to1e18(11111)
+        await staking
+          .connect(authorizer)
+          .increaseAuthorization(
+            stakingProvider.address,
+            randomBeacon.address,
+            secondIncrease
+          )
+
+        await randomBeacon
+          .connect(operator)
+          .updateOperatorStatus(operator.address)
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should have correct eligible stake", async () => {
+        expect(
+          await randomBeacon.eligibleStake(stakingProvider.address)
+        ).to.equal(initialIncrease.add(secondIncrease))
+      })
+
+      it("should have operator status updated", async () => {
+        expect(await randomBeacon.isOperatorUpToDate(operator.address)).to.be
+          .true
+      })
+    })
+
+    // Invoke `increaseAuthorization` after `authorizationDecreaseRequested`.
+    // The decrease is not yet approved when `increaseAuthorization` is called.
+    describe("authorizationDecreaseRequested -> authorizationIncreased", () => {
+      let firstDecrease
+      let secondIncrease
```

Comment:
We could make them `:number` or whatever type they have. Same comment to all other similar fields in mixed interactions section. We could also fix it in `WalletRegistry.Authorization.test.ts` in one shot.

Reply:
https://github.com/keep-network/keep-core/pull/2913